source/Octopus.Tentacle.Tests.Integration/FileTransferServiceTests.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut;
+using NUnit.Framework;
+using Octopus.Tentacle.CommonTestUtils;
+using Octopus.Tentacle.Tests.Integration.Support;
+using Octopus.Tentacle.Tests.Integration.Support.Legacy;
+using Octopus.Tentacle.Tests.Integration.Util;
+
+namespace Octopus.Tentacle.Tests.Integration
+{
+    public class FileTransferServiceTests : IntegrationTest
+    {
+        [Test]
+        [TestCaseSource(typeof(TentacleTypesToTest))]
+        public async Task UploadFileSuccessfully(TentacleType tentacleType)
+        {
+            using var fileToUpload = new RandomTemporaryFileBuilder().Build();
+
+            using var clientAndTentacle = await new ClientAndTentacleBuilder(tentacleType)
+                .Build(CancellationToken);
+
+            var uploadResult = clientAndTentacle.TentacleClient.FileTransferService.UploadFile(
+                "the_remote_uploaded_file",
+                new DataStream(
```

Comment:
It is possible to create a DataStream from a string, no need for that extra file.